src/widgets/list.rs

```diff
@@ -77,28 +84,28 @@ impl List {
         self.current_height.set(self.current_height.get() + h);
     }
 
-    // Given absolute coordinates, returns the list entry
+    // Given absolute coordinates, returns the list entry index
     // drawn at that point.
-    fn get_entry(&self, p: Point) -> Option<Arc<Entry>> {
+    fn get_entry_index(&self, p: Point) -> i32 {
```

Comment:
This function should probably return an Option<usize> or Option<u32> instead of an i32. Using None instead of -1 to indicate that there is no entry at the given point would be more idiomatic rust, I believe.